File: static/CWE-78/java/vulnerable.java
// CWE-78: OS Command Injection
// User-controlled input is passed unsanitized to an OS command execution call.
// Vulnerable sink: Runtime.getRuntime().exec(userInput)

class CWE_78 {
    public void runCommand(String userInput) throws Exception {
        Process p = Runtime.getRuntime().exec("ping -c 1 " + userInput);
        p.waitFor();
    }
}

File: static/CWE-78/csharp/vulnerable.cs
// CWE-78: Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')
// User input is appended to a shell command and executed by the OS.
// Vulnerable sink: Process.Start("cmd.exe", "/c " + userInput)

using System.Diagnostics;

namespace GuineaPig.Sast
{
    public class CWE_78
    {
        public void RunCommand(string userInput)
        {
            var psi = new ProcessStartInfo("cmd.exe", "/c " + userInput)
            {
                UseShellExecute = false,
                RedirectStandardOutput = true
            };
            Process.Start(psi);
        }
    }
}
